packages/chevrotain/src/parse/grammar/lookahead.ts

```diff
@@ -39,19 +52,30 @@ export enum PROD_TYPE {
   ALTERNATION
 }
 
-export function getProdType(prod: IProduction): PROD_TYPE {
+export function getProdType(
+  prod: IProduction | LookaheadProductionType
+): PROD_TYPE {
   /* istanbul ignore else */
-  if (prod instanceof Option) {
+  if (prod instanceof Option || prod === "Option") {
```

Comment:
why was this logic changed?
I know the `instanceof` usage can cause problems in general
But what specific problem did you encounter?

Reply:
The type of the `prod` parameter has changed to accommodate the new `LookaheadProductionType`, which is a union of strings.

Reply:
Okay, I see it serves two flows now